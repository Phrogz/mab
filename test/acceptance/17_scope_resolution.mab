setSlot( "Person", Object new )

Person setSlot( "sayHi", method(
	setSlot( "localFoo", "localFoo" )
	Number setSlot( "soundOff", method(
		p( self, self age, age, localFoo )
	) )
	p( age, self, self age, self fullName )
	self age soundOff
) )

Person setSlot( "fullName", method(
	setSlot( "myFirst", self first )
	setSlot( "myLast",  self last )
	myFirst +( " " ) +( myLast )
) )

setSlot( "gk", Person new )
gk setSlot( "first", "Gavin"   )
gk setSlot( "last",  "Kistner" )
gk setSlot( "age",   35        )

p( gk fullName, gk myFirst )
gk sayHi
p( gk fullName, gk myFirst )
