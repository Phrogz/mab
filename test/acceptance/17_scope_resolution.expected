Gavin Kistner
nil (the object)
nil (the object)
<Object instance>
35
Gavin Kistner
35
nil (the object)
nil (the object)
localFoo
Gavin Kistner
nil (the object)
